python/lsst/cp/pipe/ptc.py

```diff
@@ -367,11 +434,269 @@ def runDataRef(self, dataRef, visitPairs):
+        dataset.varModelAdu : `dict`
+            Dictionary with modeled variance in ADU^2 from data, with amp names as keys.
+
+        dataset.a00 : `dict`
+            Dictionary with a00 coefficients from full cov model in Astier+19, with amp names as keys.
+
+        daaset.b00 : `dict`
+            Dictionary with b00 coefficients from full cov model in Astier+19, with amp names as keys.
+        """
+        dataset = OutputPtcDataCovAstierDataset({}, {}, {}, {}, {}, {}, {}, {}, {}, {})
+
+        for i, amp in enumerate(covFits):
+            fit = covFits[amp]
+            meanVecFinal, varVecFinal, varVecModel, wc = fit.getNormalizedFitData(0, 0, divideByMu=False)
+            gain = fit.getGain()
+            dataset.gain.setdefault(amp, []).append(gain)
+            dataset.noise.setdefault(amp, []).append(np.sqrt(fit.getRon()))
+            dataset.signal.setdefault(amp, []).append(meanVecFinal)
+            dataset.var.setdefault(amp, []).append(varVecFinal)
+            dataset.varModel.setdefault(amp, []).append(varVecModel)
+            dataset.signalAdu.setdefault(amp, []).append(meanVecFinal/gain)
+            dataset.varAdu.setdefault(amp, []).append(varVecFinal/(gain**2))
+            dataset.varModelAdu.setdefault(amp, []).append(varVecModel/gain**2)
+            dataset.a00.setdefault(amp, []).append(fit.getA()[0, 0])
+            dataset.b00.setdefault(amp, []).append(fit.getB()[0, 0])
+
+        return dataset
+
+    def computeCovariancesAstier(self, dataRef, visitPairs, detector):
+        """Iterate over pairs of flats to calculate full covariances from the difference image.
+        This follows the treatment in Astier+19
+
+        Parameters
+        ----------
+        dataRef : list of lsst.daf.persistence.ButlerDataRef
+            dataRef for the detector for the visits to be fit.
+
+        visitPairs : `iterable` of `tuple` of `int`
+            Pairs of visit numbers to be processed together
+
+        detector : `lsst.afw.cameraGeom.Detector`
+            Detector object
+
+        Returns
+        -------
+        covariancesWithTags : `numpy.recarray`
+            Covariance measurements.
+        """
+        tupleRecords = []
+        allTags = []
+
+        self.log.info('Measuring Astier covariances using %s visits for detector %s' % (visitPairs,
+                                                                                        detector.getId()))
+
+        for (v1, v2) in visitPairs:
+            # Perform ISR on each exposure
+            dataRef.dataId['expId'] = v1
+            exp1 = self.isr.runDataRef(dataRef).exposure
+            dataRef.dataId['expId'] = v2
+            exp2 = self.isr.runDataRef(dataRef).exposure
+            del dataRef.dataId['expId']
+
+            checkExpLengthEqual(exp1, exp2, v1, v2, raiseWithMessage=True)
+            expTime = exp1.getInfo().getVisitInfo().getExposureTime()
+
+            tupleRows = []
+            for ampNumber, amp in enumerate(detector):
```

Comment:
This method duplicates most of the functionality of computeStandardPtcAndNonLinearity, with only the measurement method changing.  If the output dataset can be unified, then this code can be moved up into the run() method, with the configuration defining with measurement code is run.